### PR TITLE
Code action for creating module files.

### PR DESCRIPTION
**Pull request by Draggu**
cairo side: https://github.com/starkware-libs/cairo/pull/6720

### CODE REVIEW

File: tests/test_data/code_actions/create_module_file.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+[crate_roots]
+hello = "src"
+
+[config.global]
+edition = "2024_07"
+
+//! > cairo_code
+mo<caret>d some_mo<caret>dule;<caret>
+
+fn a<caret>(){<caret>}
```
Comment (piotmag769) [inline tests/test_data/code_actions/create_module_file.txt:16]: Doesn't test much imo, better to check cases like
```cairo
mod inline<caret>_module {
    // stuff
}
```

File: src/ide/code_actions/create_module_file.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    let item_module = db.first_ancestor_of_kind(node.clone(), SyntaxKind::ItemModule)?;
+    let item_module = ItemModule::from_syntax_node(db, item_module);
+
+    if !matches!(item_module.body(db), MaybeModuleBody::None(_)) {
+        return None;
+    }
+
+    let mut is_crate_root = false;
+
+    if let ModuleId::Submodule(submodule) = db.find_module_containing_node(&node)? {
```
Comment (piotmag769) [inline src/ide/code_actions/create_module_file.rs:24]: Does this return submodule for items like `mod xyz` in `src/lib.cairo`?

File: src/ide/code_actions/create_module_file.rs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        if matches!(submodule.parent_module(db), ModuleId::CrateRoot(_)) {
+            is_crate_root = true;
+        }
+    };
+
+    let module_name = item_module.name(db).text(db);
+    let title = format!("Create module file `{module_name}`");
+
+    let file = url.path_segments()?.last()?;
+    let extra_folder = file.strip_suffix(".cairo").unwrap_or(file).to_owned();
```
Comment (piotmag769) [inline src/ide/code_actions/create_module_file.rs:34]: [nit] could be created lower where it is used

**Review by piotmag769 — APPROVED** [review]
Wait for merge on cairo's side

File: src/ide/code_actions/create_module_file.rs (hunk 2 above)
Comment (Draggu) [reply src/ide/code_actions/create_module_file.rs:24]: Yes

**Pull request merged**